rest-service/manager_rest/test/test_plugins.py

```diff
@@ -0,0 +1,121 @@
+        plugin_id = put_plugin_response.get('id')
+        self.assertIsNotNone(plugin_id)
+        self.assertEquals(put_plugin_response.get('package_name'),
+                          TEST_PACKAGE_NAME)
+        self.assertEquals(put_plugin_response.get('package_version'),
+                          TEST_PACKAGE_VERSION)
+        get_plugin_by_id_response = self.client.plugins.get(plugin_id)
+
+        self.assertEquals(put_plugin_response, get_plugin_by_id_response)
+
+    def test_get_plugin_not_found(self):
+        try:
+            self.client.plugins.get('DUMMY_PLUGIN_ID')
+        except CloudifyClientError as e:
+            self.assertEquals('Plugin DUMMY_PLUGIN_ID not found', e.message)
+            self.assertEquals(404, e.status_code)
+
+    def test_delete_plugin(self):
+        put_response = self.upload_plugin(TEST_PACKAGE_NAME,
+                                          TEST_PACKAGE_VERSION).json
+
+        plugins_list = self.client.plugins.list()
+        self.assertEqual(1, len(plugins_list),
+                         'expecting 1 plugin result, '
+                         'got {0}'.format(len(plugins_list)))
+
+        delete_response = self.client.plugins.delete(put_response['id'])
+        self.assertEquals(put_response, delete_response)
+
+        plugins_list = self.client.plugins.list()
+        self.assertEqual(0, len(plugins_list),
+                         'expecting 0 plugin result, '
+                         'got {0}'.format(len(plugins_list)))
+
+    def test_error_delete_not_found(self):
+        try:
+            self.client.plugins.delete('none_existing_id')
+        except CloudifyClientError as e:
+            self.assertEqual('Plugin none_existing_id not found', e.message)
+
+    def test_delete_multiple_versions(self):
+        response_a = self.upload_plugin(TEST_PACKAGE_NAME,
+                                        TEST_PACKAGE_VERSION)
+        self.assertEquals('201 CREATED', response_a._status)
+        response_b = self.upload_plugin(TEST_PACKAGE_NAME,
+                                        OLD_TEST_PACKAGE_VERSION)
+        self.assertEquals('201 CREATED', response_b._status)
+
+        plugins_list = self.client.plugins.list()
+        self.assertEqual(2, len(plugins_list),
+                         'expecting 2 plugin results, '
+                         'got {0}'.format(len(plugins_list)))
+
+        delete_response = self.client.plugins.delete(response_a.json.get('id'))
+        self.assertEqual(delete_response.get('id'), response_a.json.get('id'))
+
+        plugins_list = self.client.plugins.list()
+        self.assertEqual(1, len(plugins_list),
+                         'expecting 1 plugin result, got {0}'
+                         .format(len(plugins_list)))
+
+    def test_delete_plugin_not_found(self):
+        try:
+            self.client.plugins.delete('DUMMY_PLUGIN_ID')
+        except CloudifyClientError as e:
```

Comment:
assertion of error message is IR since this is derived from the file storage manager